library/components/TDropdown.vue

```diff
@@ -43,73 +39,88 @@
             zIndex: {
                 type: String,
                 default: '999',
-            }
+            },
+			isExpanded: {
+				type: Boolean,
+				default: false,
+			}
         },
 		data: () => ({
-			internal_active: false,
 			alignClass: '',
-			popup: false,
+			isPopupOpen: false,
             componentKey: '',
 		}),
         computed: {
             activeClass() {
-                if (!this.disableHandleClass) {
-                    return this.internal_active || this.isActive ? 'border-1 active' : 'border-1 in-active'
+                if (!this.isExpanded) {
+                    return this.isPopupOpen || this.isActive ? 'border-1 active' : 'border-1 in-active'
                 }
                 return '';
             },
         },
+		created: function() {
+			document.body.addEventListener("click", this.handleOutsideClick);
+		},
+		beforeDestroy: function() {
+			document.body.removeEventListener("click", this.handleOutsideClick);
+		},
 		methods: {
 			alignPopup() {
 				const element = this.$refs.popup;
+
 				if (element) {
 					const bounding = element.getBoundingClientRect();
 					const availableWidth = window.innerWidth;
+
 					if (bounding.x < 0) {
 						return this.alignClass = 'left-0';
-					} else if (bounding.x + bounding.width > availableWidth) {
+					}
+
+					if (bounding.x + bounding.width > availableWidth) {
 						return this.alignClass = 'right-0';
 					}
 				}
+
 				return this.alignClass = '';
 			},
 			handleOutsideClick(event) {
-				const container = this.$refs.dropdownFilter;
-				if (container && event && !container.contains(event.target)) {
-					this.popup = false;
-					this.internal_active = false;
-					this.$emit('closed');
-
-					document.body.removeEventListener("click", this.handleOutsideClick);
-
-					this.$nextTick(() => {
-						this.alignPopup();
-					})
+				if (this.$refs.dropdownFilter.contains(event.target)) {
+					return;
 				}
+
+				this.isPopupOpen = false;
+				this.$emit('closed');
+				this.$nextTick(() => {
+					this.alignPopup();
+				})
 			},
 			setActiveState(e, state = true) {
-				if (!state && this.popup) return;
-				this.internal_active = state;
-				if (!state) {
-                    this.$emit('closed')
-                }
+				if (!state && this.isPopupOpen) {
+					return;
+				}
 			},
-			setPopup(toggle) {
-				this.popup = toggle;
-				this.internal_active = toggle;
-
-				if (toggle) {
-					document.body.addEventListener("click", this.handleOutsideClick);
-				} else {
-					document.body.removeEventListener("click", this.handleOutsideClick);
+			openPopup() {
```

Comment:
```suggestion
			togglePopup() {
```

Non-blocking for merging: only a suggestion to better capture that the function handles closing the popup as well as opening